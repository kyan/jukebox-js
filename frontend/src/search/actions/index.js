import Search from 'search/constants'

export const toggleSearchSidebar = (open) => {
  return {
    type: Search.TOGGLE_SEARCH_SIDEBAR,
    open
  }
}

export const storeSearchResults = (results) => {
  return {
    type: Search.STORE_SEARCH_RESULTS,
    results
  }
}

export const storeSearchQuery = (query) => {
  return {
    type: Search.STORE_SEARCH_QUERY,
    query
  }
}

export const search = (query, options) => {
  return {
    type: Search.SEARCH,
    key: Search.SEARCH_GET_TRACKS,
    params: { query, options }
  }
}

export const removeFromSearchResults = (uri) => {
  return {
    type: Search.REMOVE_FROM_SEARCH_RESULTS,
    uri
  }
}
