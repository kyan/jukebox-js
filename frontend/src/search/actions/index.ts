import Search from 'search/constants'

export const toggleSearchSidebar = (open: boolean) => {
  return {
    type: Search.TOGGLE_SEARCH_SIDEBAR,
    open
  }
}

export const storeSearchResults = (results: any[]) => {
  return {
    type: Search.STORE_SEARCH_RESULTS,
    results
  }
}

export const storeSearchQuery = (query: string | null) => {
  return {
    type: Search.STORE_SEARCH_QUERY,
    query
  }
}

export const search = (query: string, options: any) => {
  return {
    type: Search.SEARCH,
    key: Search.SEARCH_GET_TRACKS,
    params: { query, options }
  }
}

export const removeFromSearchResults = (uris: string[]) => {
  return {
    type: Search.REMOVE_FROM_SEARCH_RESULTS,
    uris
  }
}

export const addTrackToMix = (track: any) => {
  return {
    type: Search.ADD_TRACK_TO_MIX,
    track
  }
}

export const removeFromMix = (uri: string) => {
  return {
    type: Search.REMOVE_TRACK_FROM_MIX,
    uri
  }
}

export const swapTracks = (a: number, b: number) => {
  return {
    type: Search.SWAP_TRACKS,
    a,
    b
  }
}

export const clearMix = () => {
  return {
    type: Search.CLEAR_MIX
  }
}
