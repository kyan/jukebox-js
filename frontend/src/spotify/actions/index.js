import Spotify from 'spotify/constants'

export const addToken = spotifyToken => {
  return {
    type: Spotify.SPOTIFY,
    key: Spotify.ADD_TOKEN,
    params: { spotifyToken }
  }
}
