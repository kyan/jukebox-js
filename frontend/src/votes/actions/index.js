import Vote from 'votes/constants'

export const vote = (uri, vote) => {
  return {
    type: Vote.VOTE,
    key: Vote.CAST_VOTE,
    params: { uri, vote }
  }
}
