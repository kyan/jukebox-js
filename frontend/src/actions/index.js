import MopidyApi from 'constants/mopidy-api'
import Types from 'constants/common'
import { transformUrl } from 'utils/spotify'

export const updateToken = token => {
  return {
    type: Types.STORE_TOKEN,
    token
  }
}

export const clearToken = () => {
  return {
    type: Types.CLEAR_STORE_TOKEN
  }
}

export const addNewTrack = url => {
  let uri = transformUrl(url)

  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_ADD_TRACK,
    params: { 'uri': uri }
  }
}

export const addCurrentTrack = track => {
  return {
    type: Types.ADD_CURRENT_TRACK,
    track
  }
}

export const addTrackList = list => {
  return {
    type: Types.ADD_TRACKS,
    list
  }
}

export const removeFromTracklist = uri => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_REMOVE_TRACK,
    params: { 'uri': [uri] }
  }
}

export const updateProgressTimer = (position, duration) => {
  if (duration === Infinity) { duration = 0 }

  return {
    type: Types.UPDATE_PROGRESS_TIMER,
    position,
    duration
  }
}

export const wsConnect = () => {
  return {
    type: Types.CONNECT
  }
}

export const wsConnecting = () => {
  return {
    type: Types.CONNECTING
  }
}

export const wsConnected = () => {
  return {
    type: Types.CONNECTED
  }
}

export const wsDisconnect = () => {
  return {
    type: Types.DISCONNECT
  }
}

export const wsDisconnected = () => {
  return {
    type: Types.DISCONNECTED
  }
}

export const mopidyConnected = () => {
  return {
    type: Types.MOPIDY_CONNECTED
  }
}

export const mopidyDisconnected = () => {
  return {
    type: Types.MOPIDY_DISCONNECTED
  }
}

export const getCurrentTrack = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_CURRENT_TRACK
  }
}

export const getTimePosition = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_TIME_POSITION
  }
}

export const getState = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_PLAYBACK_STATE
  }
}

export const getTrackList = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_GET_TRACKS
  }
}

export const clearTrackList = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_CLEAR
  }
}

export const startPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_PLAY
  }
}

export const stopPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_STOP
  }
}

export const pausePlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_PAUSE
  }
}

export const nextPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_NEXT
  }
}

export const previousPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_BACK
  }
}

export const getVolume = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.GET_VOLUME
  }
}

export const updateVolume = (volume) => {
  return {
    type: Types.UPDATE_VOLUME,
    volume
  }
}

export const updatePlaybackState = (state) => {
  return {
    type: Types.UPDATE_PLAYBACK_STATE,
    state
  }
}

export const setVolume = (volume) => {
  return {
    type: Types.SEND,
    key: MopidyApi.SET_VOLUME,
    params: [Number(volume)]
  }
}
