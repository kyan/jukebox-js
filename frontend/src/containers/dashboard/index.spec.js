import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { Icon } from 'semantic-ui-react'
import configureMockStore from 'redux-mock-store'
import * as actions from '../../actions'
import DashboardContainer, { Dashboard } from './index'

describe('Dashboard', () => {
  let wrapper

  describe('render just the dashboard without redux', () => {
    const jukebox = { currentVolume: 25, online: true }
    const currentTrack = {}
    const tracklist = []
    const tracklistImages = {}
    const dispatchMock = jest.fn()

    it('renders as expected', () => {
      wrapper = shallow(
        <Dashboard
          jukebox={ jukebox }
          currentTrack={ currentTrack }
          tracklist={ tracklist }
          tracklistImages={ tracklistImages }
          dispatch={ dispatchMock }
        />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('componentDidMount', () => {
      spyOn(actions, 'wsConnect')
      wrapper.instance().componentDidMount()
      expect(actions.wsConnect).toHaveBeenCalled()
    })

    it('componentWillUnmount', () => {
      spyOn(actions, 'wsDisconnect')
      wrapper.instance().componentWillUnmount()
      expect(actions.wsDisconnect).toHaveBeenCalled()
    })

    it('startPlaying', () => {
      spyOn(actions, 'startPlaying')
      wrapper.instance().startPlaying()
      expect(actions.startPlaying).toHaveBeenCalled()
    })

    it('pausePlaying', () => {
      spyOn(actions, 'pausePlaying')
      wrapper.instance().pausePlaying()
      expect(actions.pausePlaying).toHaveBeenCalled()
    })

    it('skipPlaying', () => {
      spyOn(actions, 'skipPlaying')
      wrapper.instance().skipPlaying()
      expect(actions.skipPlaying).toHaveBeenCalled()
    })

    it('addNewTrack', () => {
      spyOn(actions, 'addNewTrack')
      wrapper.instance().addNewTrack('track')
      expect(actions.addNewTrack).toHaveBeenCalledWith('track')
    })

    it('onVolumeChange', () => {
      spyOn(actions, 'setVolume')
      wrapper.instance().onVolumeChange(32)
      expect(actions.setVolume).toHaveBeenCalledWith(32)
    })

    describe('handleURLDrop', () => {
      it('handles a monitor passed in', () => {
        spyOn(wrapper.instance(), 'addNewTrack')
        const monitor = {
          getItem: () => {
            return { urls: ['url123'] }
          }
        }
        wrapper.instance().handleURLDrop(null, monitor)
        expect(wrapper.instance().addNewTrack).toHaveBeenCalledWith('url123')
      })

      it('handles a monitor not passed in', () => {
        spyOn(wrapper.instance(), 'addNewTrack')
        wrapper.instance().handleURLDrop(null, null)
        expect(wrapper.instance().addNewTrack).not.toHaveBeenCalled()
      })
    })

    describe('onlineIcon', () => {
      it('handles offline', () => {
        expect(wrapper.instance().onlineIcon(false).props.color).toEqual('orange')
      })
    })
  })

  describe('render the connected app', () => {
    const store = configureMockStore()({
      tracklist: [],
      jukebox: {
        currentVolume: 25,
        online: true
      },
      timer: {
        duration: 100,
        postion: 0
      }
    })

    it('renders as expected', () => {
      wrapper = mount(
        <Provider store={store}>
          <DashboardContainer />
        </Provider>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
