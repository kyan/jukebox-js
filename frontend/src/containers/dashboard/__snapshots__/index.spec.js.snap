// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render just the dashboard without redux renders as expected 1`] = `
<DimmerDimmable
  blurring={true}
  className="jukebox-dashboard"
  dimmed={false}
>
  <Connect(Component) />
  <VolumeButtons
    onVolumeChange={[Function]}
    volume={25}
  />
  <Controls
    onNext={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onPrevious={[Function]}
  />
  <Divider />
  <Grid>
    <GridColumn
      width={6}
    >
      <DragDropContextProvider
        backend={[Function]}
      >
        <DropTarget(UrlDropArea)
          accepts={
            Array [
              "__NATIVE_URL__",
            ]
          }
          onDrop={[Function]}
        >
          <Header
            size="small"
          >
            Currently Playing
          </Header>
          <Connect(Component) />
        </DropTarget(UrlDropArea)>
      </DragDropContextProvider>
    </GridColumn>
    <GridColumn
      width={10}
    >
      <Header
        size="small"
      >
        Playlist 
        <ClearPlaylist
          onClear={[Function]}
        />
      </Header>
      <Tracklist
        currentTrack={Object {}}
        images={Object {}}
        onRemoveTrack={[Function]}
        tracks={Array []}
      />
    </GridColumn>
  </Grid>
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</DimmerDimmable>
`;

exports[`Dashboard render the connected app renders as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(DragDropContext(Dashboard))>
    <DragDropContext(Dashboard)
      dispatch={[Function]}
      jukebox={
        Object {
          "currentVolume": 25,
          "online": true,
        }
      }
      tracklist={Array []}
      tracklistImages={Object {}}
    >
      <Dashboard
        dispatch={[Function]}
        jukebox={
          Object {
            "currentVolume": 25,
            "online": true,
          }
        }
        tracklist={Array []}
        tracklistImages={Object {}}
      >
        <DimmerDimmable
          blurring={true}
          className="jukebox-dashboard"
          dimmed={false}
        >
          <div
            className="blurring dimmable jukebox-dashboard"
          >
            <Connect(Component)>
              <Component
                dispatch={[Function]}
                settings={
                  Object {
                    "open": false,
                  }
                }
              >
                <Button
                  as="button"
                  className="jb-settings-toggle"
                  floated="right"
                  icon="user"
                  onClick={[Function]}
                >
                  <button
                    className="ui icon right floated button jb-settings-toggle"
                    onClick={[Function]}
                    role="button"
                  >
                    <Icon
                      as="i"
                      key="user"
                      name="user"
                    >
                      <i
                        aria-hidden="true"
                        className="user icon"
                      />
                    </Icon>
                  </button>
                </Button>
                <Modal
                  closeIcon={true}
                  closeOnDimmerClick={true}
                  closeOnDocumentClick={false}
                  dimmer="blurring"
                  eventPool="Modal"
                  onClose={[Function]}
                  open={false}
                  size="small"
                >
                  <Portal
                    className="ui page modals dimmer transition visible active"
                    closeOnDocumentClick={false}
                    closeOnEscape={true}
                    closeOnRootNodeClick={true}
                    eventPool="Modal"
                    mountNode={<body />}
                    onClose={[Function]}
                    onMount={[Function]}
                    onOpen={[Function]}
                    onUnmount={[Function]}
                    open={false}
                    openOnTriggerClick={true}
                  />
                </Modal>
              </Component>
            </Connect(Component)>
            <VolumeButtons
              onVolumeChange={[Function]}
              volume={25}
            >
              <ButtonGroup
                floated="right"
              >
                <div
                  className="ui right floated buttons"
                >
                  <Button
                    as="button"
                    className="jb-volume-down"
                    onClick={[Function]}
                  >
                    <button
                      className="ui button jb-volume-down"
                      onClick={[Function]}
                      role="button"
                    >
                      <Icon
                        as="i"
                        name="volume down"
                      >
                        <i
                          aria-hidden="true"
                          className="volume down icon"
                        />
                      </Icon>
                    </button>
                  </Button>
                  <ButtonOr
                    text={25}
                  >
                    <div
                      className="or"
                      data-text={25}
                    />
                  </ButtonOr>
                  <Button
                    as="button"
                    className="jb-volume-up"
                    onClick={[Function]}
                  >
                    <button
                      className="ui button jb-volume-up"
                      onClick={[Function]}
                      role="button"
                    >
                      <Icon
                        as="i"
                        name="volume up"
                      >
                        <i
                          aria-hidden="true"
                          className="volume up icon"
                        />
                      </Icon>
                    </button>
                  </Button>
                </div>
              </ButtonGroup>
            </VolumeButtons>
            <Controls
              onNext={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPrevious={[Function]}
            >
              <span>
                <SkipButtons
                  onNext={[Function]}
                  onPrevious={[Function]}
                >
                  <span>
                    <Button
                      animated="vertical"
                      as="button"
                      className="jb-previous-button"
                      onClick={[Function]}
                    >
                      <button
                        className="ui vertical animated button jb-previous-button"
                        onClick={[Function]}
                        role="button"
                      >
                        <ButtonContent
                          hidden={true}
                        >
                          <div
                            className="hidden content"
                          >
                            Prev
                          </div>
                        </ButtonContent>
                        <ButtonContent
                          visible={true}
                        >
                          <div
                            className="visible content"
                          >
                            <Icon
                              as="i"
                              name="step backward"
                            >
                              <i
                                aria-hidden="true"
                                className="step backward icon"
                              />
                            </Icon>
                          </div>
                        </ButtonContent>
                      </button>
                    </Button>
                    <Button
                      animated="vertical"
                      as="button"
                      className="jb-next-button"
                      onClick={[Function]}
                    >
                      <button
                        className="ui vertical animated button jb-next-button"
                        onClick={[Function]}
                        role="button"
                      >
                        <ButtonContent
                          hidden={true}
                        >
                          <div
                            className="hidden content"
                          >
                            Next
                          </div>
                        </ButtonContent>
                        <ButtonContent
                          visible={true}
                        >
                          <div
                            className="visible content"
                          >
                            <Icon
                              as="i"
                              name="step forward"
                            >
                              <i
                                aria-hidden="true"
                                className="step forward icon"
                              />
                            </Icon>
                          </div>
                        </ButtonContent>
                      </button>
                    </Button>
                  </span>
                </SkipButtons>
                <Button
                  animated="vertical"
                  as="button"
                  onClick={[Function]}
                >
                  <button
                    className="ui vertical animated button"
                    onClick={[Function]}
                    role="button"
                  >
                    <ButtonContent
                      hidden={true}
                    >
                      <div
                        className="hidden content"
                      >
                        Play
                      </div>
                    </ButtonContent>
                    <ButtonContent
                      visible={true}
                    >
                      <div
                        className="visible content"
                      >
                        <Icon
                          as="i"
                          name="play"
                        >
                          <i
                            aria-hidden="true"
                            className="play icon"
                          />
                        </Icon>
                      </div>
                    </ButtonContent>
                  </button>
                </Button>
                <Button
                  animated="vertical"
                  as="button"
                  onClick={[Function]}
                >
                  <button
                    className="ui vertical animated button"
                    onClick={[Function]}
                    role="button"
                  >
                    <ButtonContent
                      hidden={true}
                    >
                      <div
                        className="hidden content"
                      >
                        Pause
                      </div>
                    </ButtonContent>
                    <ButtonContent
                      visible={true}
                    >
                      <div
                        className="visible content"
                      >
                        <Icon
                          as="i"
                          name="pause"
                        >
                          <i
                            aria-hidden="true"
                            className="pause icon"
                          />
                        </Icon>
                      </div>
                    </ButtonContent>
                  </button>
                </Button>
              </span>
            </Controls>
            <Divider>
              <div
                className="ui divider"
              />
            </Divider>
            <Grid>
              <div
                className="ui grid"
              >
                <GridColumn
                  width={6}
                >
                  <div
                    className="six wide column"
                  >
                    <DragDropContextProvider
                      backend={[Function]}
                    >
                      <DropTarget(UrlDropArea)
                        accepts={
                          Array [
                            "__NATIVE_URL__",
                          ]
                        }
                        onDrop={[Function]}
                      >
                        <UrlDropArea
                          accepts={
                            Array [
                              "__NATIVE_URL__",
                            ]
                          }
                          connectDropTarget={[Function]}
                          onDrop={[Function]}
                        >
                          <div>
                            <Header
                              size="small"
                            >
                              <div
                                className="ui small header"
                              >
                                Currently Playing
                              </div>
                            </Header>
                            <Connect(Component)>
                              <Component
                                dispatch={[Function]}
                                image={null}
                                progress={NaN}
                              >
                                <CurrentTrack
                                  image={null}
                                  progress={NaN}
                                />
                              </Component>
                            </Connect(Component)>
                          </div>
                        </UrlDropArea>
                      </DropTarget(UrlDropArea)>
                    </DragDropContextProvider>
                  </div>
                </GridColumn>
                <GridColumn
                  width={10}
                >
                  <div
                    className="ten wide column"
                  >
                    <Header
                      size="small"
                    >
                      <div
                        className="ui small header"
                      >
                        Playlist 
                        <ClearPlaylist
                          onClear={[Function]}
                        >
                          <span>
                            <Label
                              as="a"
                              className="jb-clear-button"
                              color="red"
                              horizontal={true}
                              onClick={[Function]}
                              size="mini"
                            >
                              <a
                                className="ui red mini horizontal label jb-clear-button"
                                onClick={[Function]}
                              >
                                CLEAR
                              </a>
                            </Label>
                            <Confirm
                              cancelButton="No thanks"
                              confirmButton="Do it!"
                              content="Are you sure you want to nuke the playlist?"
                              onCancel={[Function]}
                              onConfirm={[Function]}
                              open={false}
                            >
                              <Modal
                                closeOnDimmerClick={true}
                                closeOnDocumentClick={false}
                                dimmer={true}
                                eventPool="Modal"
                                onClose={[Function]}
                                open={false}
                                size="small"
                              >
                                <Portal
                                  className="ui page modals dimmer transition visible active"
                                  closeOnDocumentClick={false}
                                  closeOnEscape={true}
                                  closeOnRootNodeClick={true}
                                  eventPool="Modal"
                                  mountNode={<body />}
                                  onClose={[Function]}
                                  onMount={[Function]}
                                  onOpen={[Function]}
                                  onUnmount={[Function]}
                                  open={false}
                                  openOnTriggerClick={true}
                                />
                              </Modal>
                            </Confirm>
                          </span>
                        </ClearPlaylist>
                      </div>
                    </Header>
                    <Tracklist
                      images={Object {}}
                      onRemoveTrack={[Function]}
                      tracks={Array []}
                    >
                      <List
                        relaxed={true}
                      >
                        <div
                          className="ui relaxed list"
                          role="list"
                        />
                      </List>
                    </Tracklist>
                  </div>
                </GridColumn>
              </div>
            </Grid>
            <ToastContainer
              autoClose={5000}
              bodyClassName=""
              className={null}
              closeButton={
                <DefaultCloseButton
                  ariaLabel="close"
                />
              }
              closeOnClick={true}
              hideProgressBar={false}
              newestOnTop={false}
              pauseOnHover={true}
              position="top-right"
              progressClassName=""
              style={null}
              toastClassName=""
              transition={[Function]}
            >
              <div />
            </ToastContainer>
          </div>
        </DimmerDimmable>
      </Dashboard>
    </DragDropContext(Dashboard)>
  </Connect(DragDropContext(Dashboard))>
</Provider>
`;
